packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/plugin-ledger-connector-iroha.ts

```diff
@@ -0,0 +1,684 @@
+    return this.instanceId;
+  }
+
+  public async onPluginInit(): Promise<unknown> {
+    return;
+  }
+
+  public getHttpServer(): Optional<Server | SecureServer> {
+    return Optional.ofNullable(this.httpServer);
+  }
+
+  public async shutdown(): Promise<void> {
+    const serverMaybe = this.getHttpServer();
+    if (serverMaybe.isPresent()) {
+      const server = serverMaybe.get();
+      await promisify(server.close.bind(server))();
+    }
+  }
+
+  async registerWebServices(
+    app: Express,
+    //wsApi: SocketIoServer,
+  ): Promise<IWebServiceEndpoint[]> {
+    //const { web3 } = this;
+    //const { logLevel } = this.options;
+    const webServices = await this.getOrCreateWebServices();
+    await Promise.all(webServices.map((ws) => ws.registerExpress(app)));
+
+    // wsApi.on("connection", (socket: SocketIoSocket) => {
+    //   this.log.debug(`New Socket connected. ID=${socket.id}`);
+
+    //   socket.on(WatchBlocksV1.Subscribe, () => {
+    //     new WatchBlocksV1Endpoint({ web3, socket, logLevel }).subscribe();
+    //   });
+    // });
+    return webServices;
+  }
+
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    if (Array.isArray(this.endpoints)) {
+      return this.endpoints;
+    }
+    const endpoints: IWebServiceEndpoint[] = [];
+    {
+      const endpoint = new RunTransactionEndpoint({
+        connector: this,
+        logLevel: this.options.logLevel,
+      });
+      endpoints.push(endpoint);
+    }
+    // {
+    //   const endpoint = new InvokeContractEndpoint({
+    //     connector: this,
+    //     logLevel: this.options.logLevel,
+    //   });
+    //   endpoints.push(endpoint);
+    // }
+
+    {
+      const opts: IGetPrometheusExporterMetricsEndpointV1Options = {
+        connector: this,
+        logLevel: this.options.logLevel,
+      };
+      const endpoint = new GetPrometheusExporterMetricsEndpointV1(opts);
+      endpoints.push(endpoint);
+    }
+    this.endpoints = endpoints;
+    return endpoints;
+  }
+
+  public getPackageName(): string {
+    return `@hyperledger/cactus-plugin-ledger-connector-iroha`;
+  }
+
+  public async getConsensusAlgorithmFamily(): Promise<
+    ConsensusAlgorithmFamily
+  > {
+    return ConsensusAlgorithmFamily.Authority;
+  }
+  public async hasTransactionFinality(): Promise<boolean> {
+    const currentConsensusAlgorithmFamily = await this.getConsensusAlgorithmFamily();
+
+    return consensusHasTransactionFinality(currentConsensusAlgorithmFamily);
+  }
+  //...  spread
+  //pass in the params
+  public async transact(
+    req: RunTransactionRequest, //string + array<any>
+  ): Promise<RunTransactionResponse> {
+    const { baseConfig } = req;
+    if (
+      !baseConfig ||
+      !baseConfig.privKey ||
+      !baseConfig.creatorAccountId ||
+      !baseConfig.irohaHost ||
+      !baseConfig.irohaPort ||
+      !baseConfig.quorum ||
+      !baseConfig.timeoutLimit
+    ) {
+      console.error("No data here!");
```

Comment:
@hxlaf 

1. Please use the logger object instead of the built-in console log APIs.
2. Please log this on a debug level, invalid user input (e.g. missing parameters) are not an error on the back-end side. A simple test to figure out what should be logged as an error: If this log message should be investigated as a bug, then it should be on the error level.

Reply:
Applied this.log.debug('message')! Thank you!
![image](https://user-images.githubusercontent.com/10954157/129980940-dedf2db5-1071-43c0-a5f2-bbc65b6a5dbd.png)